crates/ide/src/syntax_highlighting/inject.rs

```diff
@@ -243,10 +246,10 @@ pub(super) fn doc_comment(hl: &mut Highlights, sema: &Semantics<RootDatabase>, n
         }
     }
 
-    for range in intra_doc_links {
+    for (def, range) in intra_doc_links {
         hl.add(HlRange {
             range,
-            highlight: HlTag::IntraDocLink | HlMod::Documentation,
+            highlight: module_def_to_hl_tag(def) | HlMod::Documentation | HlMod::IntraDocLink,
```

Comment:
Should this also get the `Injected` Modifier? I'm unsure what exactly injection refers to, is it only source code highlighting or more general?